lib/hardware.go

```diff
@@ -0,0 +1,47 @@
+package lib
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	"github.com/cybozu-go/well"
+)
+
+func DetectCPUNumber() (int, error) {
```

Comment:
Just as a matter of taste, I prefer `CountCPUs` and `CountMemoryModules`.
`DetectCPUNumber` sounds like detecting the sequential number of the specified CPU.